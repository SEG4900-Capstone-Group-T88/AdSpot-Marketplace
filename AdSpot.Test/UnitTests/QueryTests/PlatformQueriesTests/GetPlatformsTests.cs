using AdSpot.Models;

namespace AdSpot.Test.UnitTests.PlatformQueriesTests;

[Collection("adspot-inmemory-db")]
public class GetPlatformsTests 
{
    public const string GetAllPlatformsQuery = """
        query GetPlatforms{
            platforms {
                platformId
                name
            }
        }
        """;

    //[Fact]
    [Trait("Category", "Unit")]
    public async Task GetPlatformsSuccessful()
    {
        var result = await TestServices.ExecuteRequestAsync(b =>
            b.SetQuery(GetAllPlatformsQuery)
        );

        result.MatchSnapshot();
    }
}