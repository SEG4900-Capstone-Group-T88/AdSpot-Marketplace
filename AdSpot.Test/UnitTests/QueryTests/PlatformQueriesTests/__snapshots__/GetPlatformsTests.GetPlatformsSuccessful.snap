{
  "data": {
    "platforms": [
      {
        "platformId": 1,
        "name": "Facebook"
      },
      {
        "platformId": 2,
        "name": "Twitter"
      },
      {
        "platformId": 3,
        "name": "Instagram"
      },
      {
        "platformId": 4,
        "name": "Youtube"
      }
    ]
  }
}